File: serverApp/src/main/java/com/server/dto/GuideDTO.java
package com.server.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class GuideDTO {

    private Long id;

    private String creatorLogin;

    private String title;

    private String fileBytes;

    private Timestamp editDate;

    private Boolean isBlocked;

    private Boolean addedToFavorites;
}

File: guide_app/lib/common/dto/guide_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'guide_dto.g.dart';

/// Guide dto.
/// For code generation run  flutter pub run build_runner build --delete-conflicting-outputs.
@JsonSerializable()
class GuideDto {
  GuideDto(this.id, this.login, this.title, this.content, this.editDate,
      this.isBlocked);

  final int id;

  @JsonKey(name: 'creatorLogin')
  final String login;

  final String title;

  // Quill delta in json format.
  @JsonKey(name: 'fileBytes')
  final String content;

  final DateTime editDate;

  final bool isBlocked;

  factory GuideDto.fromJson(Map<String, dynamic> json) =>
      _$GuideDtoFromJson(json);
}
